demo/test.c

```diff
@@ -2214,6 +2214,50 @@ static int test_s_mp_toom_sqr(void)
    return EXIT_FAILURE;
 }
 
+static int test_mp_pack_unpack(void)
+{
+   mp_int a, b;
+   int err;
+   size_t size, countp;
+   unsigned char *buf = NULL;
+
+   mp_order order = MP_LSB_FIRST;
+   mp_endian endianess = MP_NATIVE_ENDIAN;
+
+   if ((err = mp_init_multi(&a, &b, NULL)) != MP_OKAY)                       goto LTM_ERR;
+   if ((err = mp_rand(&a, 15)) != MP_OKAY)                                   goto LTM_ERR;
+
+   size = (size_t)(((mp_count_bits(&a) + 1) + 7) / 8);
+
+   buf = MP_MALLOC(size);
+   if (buf == NULL) {
+      fprintf(stderr, "test_pack_unpack failed to allocate %zu bytes\n",
+              sizeof(*buf) * size);
+      goto LTM_ERR;
+   }
+
+   if ((err = mp_pack((void *)buf, &countp, order, 1,
+                      endianess, 0, &a, size)) != MP_OKAY)                   goto LTM_ERR;
+   if ((err = mp_unpack(&b, countp, order, 1,
+                        endianess, 0, (const void *)buf)) != MP_OKAY)        goto LTM_ERR;
+
+   if (mp_cmp(&a, &b) != MP_EQ) {
+      fprintf(stderr, "pack/unpack cycle failed\n");
+      goto LTM_ERR;
+   }
+
+   free(buf);
```

Comment:
Free or mp_free? Is this consistent in the test suite

Reply:
> Is this consistent in the test suite
(I'll take it as a question) The rest of the tests is done with static buffers didn't want to do it here, although it is possible.